smurfs/src/components/SmurfForm.js

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import Smurf from "./Smurf";
+import { getSmurfs, addSmurf, deleteSmurf } from "../actions/index";
+
+class SmurfForm extends Component {
+  state = {
+    newSmurf: {
+      name: "",
+      age: "",
+      height: ""
+    }
+  };
+
+  componentDidMount() {
+    this.props.getSmurfs();
+  }
+
+  changeHandler = e => {
+    this.setState({
+      newSmurf: {
+        ...this.state.newSmurf,
+        [e.target.name]: e.target.value
+      }
+    });
+  };
+
+  submitHandler = e => {
+    e.preventDefault();
+    this.props.addSmurf(this.state.newSmurf);
+    this.setState({
+      newSmurf: {
+        name: "",
+        age: "",
+        height: ""
+      }
+    });
+  };
+
+  render() {
+    return (
+      <div className="App">
```

Comment:
this is fine the way it is; however, I would probably separate the list and the form into different components if this were a real-life scenario. But what you did here works.